lotchay/CS-5780-SimplifiedSSL
# Implement the SSL layer

What
- Added client and server interactions.
- Added SSL handshake, and encoded data transfer.
- Fixed RSA to cast unsigned byte array to big integer (which is signed).

How to test
- Clone this repo
- Install Maven locally
- Run `mvn clean install` at the root of this project
- Navigate to target directory
- Run `jar -xvf simplifiedssl.jar`
- Open another terminal and navigate to the same target directory
- Run `java  -Dserver.private_key=private_key.txt    -Dserver.users=users.txt -Dserver.port=3445  Server` on one terminal
- Run `java Client localhost 3445 mickey < users.txt` on the other terminal
- Edit users.txt to test

Check
- Output should be relatively close to the provided model solution.
